### PR TITLE
Adding KUTTL tests for manila operator

### PR DESCRIPTION
**Pull request by vhariria**
Add KUTTL tests for Manila operator

### CODE REVIEW

**Comment by abays** [comment]
Three of the four files changed in this PR are causing the CI precommit check to fail due to:

```
\ No newline at end of file
```

File: tests/kuttl/common/assert_sample_deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+   manilaScheduler:
+      replicas: 1
+      containerImage: quay.io/tripleozedcentos9/openstack-manila-scheduler:current-tripleo
+   manilaBackup:
+      replicas: 1
+      containerImage: quay.io/tripleozedcentos9/openstack-manila-backup:current-tripleo
+   manilaVolumes:
+      volume1:
+         containerImage: quay.io/tripleozedcentos9/openstack-manila-volume:current-tripleo
+         replicas: 1
```
Comment (abays) [inline tests/kuttl/common/assert_sample_deployment.yaml:26]: These `manilaBackup` and `manilaVolumes` blocks do not seem to exist in the actual `Manila` CRD.  Was this copy-pasted from perhaps `Cinder` KUTTL tests and then not modified?  It appears that way to me.

Comment (vhariria) [reply tests/kuttl/common/assert_sample_deployment.yaml]: Thanks for your feedback, resolved.

File: tests/kuttl/tests/manila_scale/01-assert.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+../../common/assert_sample_deployment.yaml
```
Comment (frenzyfriday) [inline tests/kuttl/tests/manila_scale/01-assert.yaml:1]: This has to be a symlink (it looks the same in github after comit though)
Example: https://github.com/openstack-k8s-operators/cinder-operator/pull/119/files#diff-914634bb88a93bef3c030b81fdd8c691a91f20c1478c9acecf4879fa2daf692c - 01-assert.yaml has the symlink folder sign

Comment (frenzyfriday) [inline tests/kuttl/tests/manila_scale/01-assert.yaml:1]: This has to be a symlink (it looks the same in github after comit though)
Example: https://github.com/openstack-k8s-operators/cinder-operator/pull/119/files#diff-914634bb88a93bef3c030b81fdd8c691a91f20c1478c9acecf4879fa2daf692c - 01-assert.yaml has the symlink folder sign

**Review by frenzyfriday — CHANGES_REQUESTED** [review]

File: tests/kuttl/common/assert_sample_deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                 - /usr/local/bin/kolla_set_configs && /usr/local/bin/kolla_start
+              command:
+                 - /bin/bash
+              image: quay.io/tripleozedcentos9/openstack-manila-api:current-tripleo
+              imagePullPolicy: IfNotPresent
+              livenessProbe:
+                 failureThreshold: 3
+                 httpGet:
+                    path: /healthcheck
+                    port: 8776
```
[the remark this answers is not in the document]
Comment (viroel) [reply tests/kuttl/common/assert_sample_deployment.yaml:74]: Comparing with [1]

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38393/rehearse-38393-pull-ci-openstack-k8s-operators-manila-operator-main-manila-operator-build-deploy/1647915011059748864/artifacts/manila-operator-build-deploy/openstack-k8s-operators-gather/artifacts/deployment.apps/manila.yaml

File: tests/kuttl/common/assert_sample_deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+                 initialDelaySeconds: 5
+                 periodSeconds: 5
+                 successThreshold: 1
+                 timeoutSeconds: 5
+              resources: {}
+              securityContext:
+                 runAsUser: 0
+              terminationMessagePath: /dev/termination-log
+              terminationMessagePolicy: File
+              volumeMounts:
```
Comment (viroel) [inline tests/kuttl/common/assert_sample_deployment.yaml:96]: We probably need to add mountPath: /etc/ceph here too

**Review by viroel — CHANGES_REQUESTED** [review]

**Pull request closed without merging**